src/transitions/Grow.js

```diff
@@ -0,0 +1,172 @@
+// @flow
+
+import React, { Component } from 'react';
+import type { Element } from 'react';
+import withTheme from '../styles/withTheme';
+import Transition from '../internal/Transition';
+import type { TransitionCallback } from '../internal/Transition';
+
+export function getScale(value: number) {
+  return `scale(${value}, ${value ** 2})`;
+}
+
+type DefaultProps = {
+  theme: Object,
+  transitionDuration: 'auto',
+};
+
+export type Props = {
+  /**
+   * The content of the component.
+   */
+  children?: Element<*>,
+  /**
+   * Callback fired before the component is entering
+   */
+  onEnter?: TransitionCallback,
+  /**
+   * Callback fired when the component is entering
+   */
+  onEntering?: TransitionCallback,
+  /**
+   * Callback fired when the component has entered
+   */
+  onEntered?: TransitionCallback, // eslint-disable-line react/sort-prop-types
+  /**
+   * Callback fired before the component is exiting
+   */
+  onExit?: TransitionCallback,
+  /**
+   * Callback fired when the component is exiting
+   */
+  onExiting?: TransitionCallback,
+  /**
+   * Callback fired when the component has exited
+   */
+  onExited?: TransitionCallback, // eslint-disable-line react/sort-prop-types
+  /**
+   * Use that property to pass a ref callback to the root component.
+   */
+  rootRef?: Function,
+  /**
+   * @ignore
+   */
+  theme?: Object,
```

Comment:
It's not required externally, but it is internally, that's why it's optional in `Props` and required in `DefaultProps`